CIP-?/README.md

```diff
@@ -0,0 +1,783 @@
+we would need to specify what 'length' we assume they have, and some policy
+of 'padding' when the length requested is longer than one, or both,
+arguments. This feels unnecessary, and it isn't even clear exactly how we
+should do this: for example, how would negative numbers be padded?
+
+These complexities, and many more besides, are poor choices, owing more to the
+legacy of C than any real useful functionality. Furthermore, they feel like a
+casual and senseless undermining of type safety and its guarantees for very
+small and questionable gains. Therefore, defining bitwise operations on
+`BuiltinInteger` is not something we wish to support.
+
+There are legitimate cases where a conversion from `BuiltinInteger` to
+`BuiltinByteString` is desirable; this conversion should be provided, and
+be both explicit and specified in a way that is independent of the machine or
+the implementation of `BuiltinInteger`, as well as total and
+round-tripping. Arguably, it is also desirable to provide built-in support for
+`BuiltinByteString` literals specified in a way convenient to their
+treatment as blobs of bytes (for example, hexadecimal or binary notation), but
+this is outside the scope of this proposal.
+
+# Specification
+
+## Proposed operations
+
+We propose several classes of operations. Firstly, we propose two operations for
+inter-conversion between  `BuiltinByteString` and `BuiltinInteger`:
+
+```haskell
+integerToByteString :: BuiltinInteger -> BuiltinByteString
+```
+Convert a number to a bitwise representation.
+
+---
+```haskell
+byteStringToInteger :: BuiltinByteString -> BuiltinInteger
+```
+Reinterpret a bitwise representation as a number.
+
+---
+We also propose several logical operations on `BuiltinByteString`s:
+
+```haskell
+andByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical AND on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+iorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical IOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+xorByteString :: BuiltinByteString -> BuiltinByteString -> BuiltinByteString
+```
+Perform a bitwise logical XOR on arguments of the same
+length, producing a result of the same length, erroring otherwise.
+
+---
+```haskell
+complementByteString :: BuiltinByteString -> BuiltinByteString
+```
+Complement all the bits in the argument, producing a
+result of the same length.
+
+---
+Lastly, we define the following additional operations:
+
+```haskell
+shiftByteString :: BuiltinByteString -> BuiltinInteger -> BuiltinByteString
+```
+Performs a bitwise shift of the first argument by a number of bit positions
+equal to the absolute value of the second argument, the direction of the shift
+being indicated by the sign of the second argument.
```

Comment:
This still doesn't say _how_ the direction is indicated!

Reply:
Clarified in the text, also for rotations.